modules/core/src/main/java/org/apache/synapse/config/SynapseConfigUtils.java

```diff
@@ -895,6 +897,23 @@ public static boolean isFailSafeEnabled(String componentName) {
         return false;
     }
 
+    public static String processString(String input, Pattern pattern, MessageContext messageContext) {
+        Matcher matcher = pattern.matcher(input);
+        StringBuffer result = new StringBuffer();
```

Comment:
Any particular reason for using StringBuffer over StringBuilder?

Reply:
Updated using StringBuilder